e2e/imgbuild/imgbuild.go

```diff
@@ -1530,43 +1530,96 @@ echo 'export LEGACY_TEST_ENV=legacy-value' >> $SINGULARITY_ENVIRONMENT
 	)
 }
 
+func (c *imgBuildTests) testContainerBuildUnderFakerootModes(t *testing.T) {
+	tmpDir, cleanup := e2e.MakeTempDir(t, c.env.TestDir, "test-container-build-under-fakeroot-modes-", "")
+	defer cleanup(t)
+
+	// running under the mode 1, 1a (--with-suid) (https://apptainer.org/docs/user/main/fakeroot.html)
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--force", "--fakeroot", fmt.Sprintf("%s/openssh-mode1a.sif", tmpDir), "testdata/unprivileged_build.def"),
+		e2e.ExpectExit(0),
+	)
+
+	// running under the mode 1, 1b (--without-suid) (https://apptainer.org/docs/user/main/fakeroot.html)
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--force", "--fakeroot", "--userns", fmt.Sprintf("%s/openssh-mode1b.sif", tmpDir), "testdata/unprivileged_build.def"),
+		e2e.ExpectExit(0),
+	)
+
+	// running under the mode 2(https://apptainer.org/docs/user/main/fakeroot.html)
+	c.env.RunApptainer(
+		t,
+		e2e.WithProfile(e2e.UserNamespaceProfile),
+		e2e.WithCommand("build"),
+		e2e.WithArgs("--force", "--userns", "--ignore-subuid", "--ignore-fakeroot-command", fmt.Sprintf("%s/openssh-mode2.sif", tmpDir), "testdata/unprivileged_build.def"),
+		e2e.ExpectExit(255), // because chown will fail
```

Comment:
It's correct that we expect this to fail, but I would like to see an additional test where this succeeds, one that installs a different package.  For example it could install epel-release.

Reply:
@DrDaveD all comments are addressed in this commit
https://github.com/apptainer/apptainer/pull/607/commits/8d826328547cae7926ea93461896d481a7c166d7